sympy/parsing/c/c_parser.py

```diff
@@ -236,20 +238,55 @@ def transform_var_decl(self, node):
                 # List in case of variable assignment, FunctionCall node in case of a funcion call
                 if (child.kind == cin.CursorKind.INTEGER_LITERAL
                     or child.kind == cin.CursorKind.UNEXPOSED_EXPR):
-                    if (node.type.kind == cin.TypeKind.INT):
-                        type = IntBaseType(String('integer'))
-                        value = Integer(val)
-                    elif (node.type.kind == cin.TypeKind.FLOAT):
-                        type = FloatBaseType(String('real'))
-                        value = Float(val)
+                    # when only one unexposed_expr is assigned e.g, b=a
```

Comment:
How about 

```python
if node.type.kind == ..INT:
    type = IntBaseType(String('integer'))
    if isinstance(val, str):
        value = val
    elif isinstance ...
        value = ..
elif node.type.kind == ..FLOAT:
    ...
else
    raise NotImplementedError()
return ...
```

That might reduce the redundancy a little

Reply:
Good idea!

Reply:
Does it look good now?